### PR TITLE
Feature/project description

### PR DESCRIPTION
**Pull request by robert-ulbrich-mercedes-benz**
## Why are the changes needed?

This change allows to set the description of a Flyte project. It could hold the responsible team for a Flyte project including the contact data or some other information. 

## What changes were proposed in this pull request?

* A new config map file was introduced including keys in the Helm chart were updated for the `flyte-core` Helm chart so they can hold a name and a description for the project
* A new structure was introduced that holds the project information to be handed over to the database repository
* For the `flyte-binary` and the single binary approach the structure of the Helm chart was changed

## How was this patch tested?

* The implementation was tested by running the regular set up process for the single binary approach
* Also the regular `flyteadmin migrate seed-projects` was tested with the adapted `flyte-single-binary-local.yaml` config file

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Docs link

The program was tested solely for our own use cases, which might differ from yours

Robert Ulbrich [robert.ulbrich@mercedes-benz.com](mailto:robert.ulrbich@mercedes-benz.com), Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: .run/single-binary.run.xml [hunk 1]
```diff
@@ -10,4 +10,4 @@
     <filePath value="$PROJECT_DIR$" />
     <method v="2" />
   </configuration>
-</component>
\ No newline at end of file
+</component>
```
Comment (LucasZanellaMBTI) [inline .run/single-binary.run.xml]: Where does this change come from?

File: cmd/single/start.go [hunk 2]
```diff
@@ -71,9 +72,11 @@ func startAdmin(ctx context.Context, cfg Admin) error {
...
 
 	logger.Infof(ctx, "Seeding default projects...")
-	projects := []string{"flytesnacks"}
-	if len(cfg.SeedProjects) != 0 {
-		projects = cfg.SeedProjects
+	projects := []config.Project{{Name: "flytesnacks", Description: "flytesnacks project"}}
+	if len(cfg.SeedProjects) > 0 {
+		for _, project := range cfg.SeedProjects {
+			projects = append(projects, config.Project{Name: project.Name, Description: project.Description})
+		}
```
Comment (LucasZanellaMBTI) [inline cmd/single/start.go:79]: AFAIK doing it this way would keep project `flytesnacks` while it was dropped in favor of configured projects previoulsy

Comment (LucasZanellaMBTI) [inline cmd/single/start.go]: AFAIK it will keep "flytesnacks" even if someone defines his/her own projects, right? Previoulsy it wasn't the case

Comment (robert-ulbrich-mercedes-benz) [reply cmd/single/start.go]: You are right, I have fixed it

File: .run/single-binary.run.xml (hunk 1 above)
Comment (robert-ulbrich-mercedes-benz) [reply .run/single-binary.run.xml]: Automatically changed from IntelliJ

File: flyteadmin/auth/authzserver/token_test.go [hunk 3]
```diff
@@ -73,7 +73,7 @@ func Test_tokenEndpoint(t *testing.T) {
 
 		m := map[string]interface{}{}
 		assert.NoError(t, json.Unmarshal(rw.Body.Bytes(), &m))
-		assert.Equal(t, 0.5*time.Hour.Seconds()-1, m["expires_in"])
+		assert.GreaterOrEqual(t, m["expires_in"], 0.5*time.Hour.Seconds()-5)
```
Comment (LucasZanellaMBTI) [inline flyteadmin/auth/authzserver/token_test.go:76]: Personally I would separate this using a dedicated fix but not sure if flyte guys care about

**Review by LucasZanellaMBTI — APPROVED** [review]
Good job! 

Please consider simplifying your feature branch by dropping merge commits.

**Comment by LucasZanellaMBTI** [comment]
@robert-ulbrich-mercedes-benz  please could you get rid of merge commits and squash changes into a single commit

**Comment by robert-ulbrich-mercedes-benz** [comment]
> Good job!
> 
> Please consider simplifying your feature branch by dropping merge commits.

I rebased the commits

**Comment by LucasZanellaMBTI** [comment]
Please add required footer sections to PR description as outlined [here](https://fosswik-confluence.dot.app.corpintra.net/display/0064/Example+Contribution?src=contextnavpagetreemode)

**Comment by robert-ulbrich-mercedes-benz** [comment]
> Please add required footer sections to PR description as outlined [here](https://fosswik-confluence.dot.app.corpintra.net/display/0064/Example+Contribution?src=contextnavpagetreemode)

What exactly do you mean? I will close this pull request. I will not further follow it, but open a pull request against flyte.

**Comment by LucasZanellaMBTI** [comment]
Completely aware of it. Nevertheless I'd prefer to have as much as possible being reviewed upfront. At least for me, legal notices are worth to be reviewed as well.

**Comment by robert-ulbrich-mercedes-benz** [comment]
> Completely aware of it. Nevertheless I'd prefer to have as much as possible being reviewed upfront. At least for me, legal notices are worth to be reviewed as well.

I added the the legal notices.

**Comment by LucasZanellaMBTI** [comment]
> > Completely aware of it. Nevertheless I'd prefer to have as much as possible being reviewed upfront. At least for me, legal notices are worth to be reviewed as well.
> 
> I added the the legal notices.

I think in PR desc follwing phrase is misssing: 
`The program was tested solely for our own use cases, which might differ from yours`

**Comment by robert-ulbrich-mercedes-benz** [comment]
> > > Completely aware of it. Nevertheless I'd prefer to have as much as possible being reviewed upfront. At least for me, legal notices are worth to be reviewed as well.
> > 
> > 
> > I added the the legal notices.
> 
> I think in PR desc follwing phrase is misssing: `The program was tested solely for our own use cases, which might differ from yours`

I adde the phrase.

**Pull request closed without merging**